app/controllers/bike_share_app.rb

```diff
@@ -1,3 +1,131 @@
 class BikeShareApp < Sinatra::Base
 
+  get '/' do
+    erb :index
+  end
+  
+  get "/stations" do
+    @stations = Station.all
+    erb :"stations/index"
+  end
+
+  get "/stations/new" do
+    erb :"stations/new"
+  end
+
+  get "/stations/:id/edit" do
+    @station = Station.find(params[:id])
+    erb :"stations/edit"
+  end
+
+  get "/stations/:id" do
+    @station = Station.find(params[:id])
+    erb :"stations/show"
+  end
+
+  post "/stations" do
+    Station.create(params[:station])
+    redirect "/stations"
+  end
+
+  delete "/stations/:id" do
+    Station.destroy(params[:id])
+    redirect "/stations"
+  end
+
+  put "/stations/:id" do
+    Station.update(params[:id], params[:station])
+    redirect "/stations"
+  end
+
+  get "/station-dashboard" do
+    erb :"stations/dashboard"
+  end
+
+  get "/condition-dashboard" do
+    if params.empty?
+      @dynamic = nil
+      @params = nil
+    else
+      @dynamic = Condition.ranges_with_trips(params[:attribute].to_sym, params[:increment].to_f)
+      @params = params
+    end
+    erb :"conditions/dashboard"
+  end
+
+  get "/trip-dashboard" do
+    erb :"trips/dashboard"
+  end
+
+  get "/trips" do
+    @number = params[:number].to_i
+    start = @number * 29 + 1
+    finish = start + 29
+    @current_trips = Trip.order(:start_date).reverse_order.where( :id => [start .. finish] )
```

Comment:
this is a lot of logic in your controller. we could move it to a PORO or possibly another method in your `Trip` model to take care of this logic